client/ayon_blender/plugins/publish/validate_absolute_external_data_paths.py

```diff
@@ -0,0 +1,76 @@
+import bpy
+import pyblish.api
+from ayon_core.pipeline import (
+    OptionalPyblishPluginMixin
+)
+from ayon_core.pipeline.publish import (
+    RepairContextAction,
+    ValidateContentsOrder,
+    PublishValidationError
+)
+
+
+class ValidateAbsoluteDataBlockPaths(pyblish.api.ContextPlugin,
+                                     OptionalPyblishPluginMixin):
+    """Validates Absolute Data Block Paths
+
+    This validator checks if all external data paths are absolute
+    to ensure the links would not be broken when publishing
+    """
+
+    label = "Validate Absolute Data Block Paths"
+    order = ValidateContentsOrder
+    hosts = ["blender"]
+    families = ["workfile"]
+    optional = True
+    actions = [RepairContextAction]
+
+    @classmethod
+    def get_invalid(cls, context):
+        invalid = []
+        object_type = type(bpy.data.objects)
+        for attr in dir(bpy.data):
+            collections = getattr(bpy.data, attr)
+            if not isinstance(collections, object_type):
+                continue
+            for data_block in collections:
+                if not hasattr(data_block, "filepath"):
+                    continue
+                if not data_block.filepath:
+                    continue
+                if data_block.filepath == bpy.path.abspath(data_block.filepath):
+                    continue
+
+                cls.log.error(f"Data block filepath {data_block.filepath} "
+                              "is not absolute path")
+                invalid.append(data_block.filepath)
+        return invalid
+
+    def process(self, context):
+        if not self.is_active(context.data):
+            self.log.debug("Skipping Validate Absolute Data Block Paths...")
+            return
+        invalid = self.get_invalid(context)
+        if invalid:
+            raise PublishValidationError(
+                "Invalid Data block filepaths",
+                title="Relative Data block filepaths",
+                description=self.get_description()
+            )
+
+    @classmethod
+    def get_description(cls):
+        return ("""
+            ### Data block filepaths are invalid
+            Data block filepaths must be absolute paths to avoid issues during relocation
+            of the published workfile into the publish folder.
+
+            #### How to repair?
+
+            Using the Repair action will turn all datablock filepaths in your scene into
+            absolute filepaths.
+        """)
```

Comment:
Does this work nicely out of the box? I'd have expected this would need a `inspect.cleandocs()` call as well to avoid the indentations of the string being problematic in the markdown formatting.

Reply:
I dont think it has been fixed at all without insepct.cleandocs()
![image](https://github.com/user-attachments/assets/35c56187-4804-4a9a-bd41-2c6d571041c3)
With the inspect.cleandocs
![image](https://github.com/user-attachments/assets/c28fb35b-1690-4c93-89c2-928454473e48)